### ISSUE TITLE
Undocumented pair pos syntax?

### ISSUE THREAD

**Issue by cmyr** [adobe-type-tools/afdko#1757]
According to the spec GPOS pair positioning rules can be expressed as either,

```
# format A
position <glyph|glyphclass> <valuerecord> <glyph|glyphclass> <valuerecord>;

# or format B
position <glyph|glyphclass> <glyph|glyphclass> <valuerecord>;
```

but it looks like the compiler will also accept,

```
position <glyph|glyphclass> <valuerecord> <glyph|glyphclass>;
```

which compiles identically to format B.

Semantically this feels like "format A with an implicitly NULL second value record", and that's reasonable if redundant, but the docs should be updated to reflect that this is allowed, if it is supposed to be allowed?

**Comment by frankrolf** [comment]
The context of a GPOS pair be more than two items – e.g. https://github.com/adobe-fonts/source-serif/blob/main/Roman/Instances/Text/Regular/kern_ctxt.fea

Maybe this makes more sense in that context?

**Comment by anthrotype** [comment]
I don't think Colin is referring to contextual positioning. This is simple (non contextual) GPOS pair positioning in which the second valuerecord gets omitted (impliclitly NULL).

**Comment by frankrolf** [comment]
FWIW, every kerning pair is contextual – unless you adjust a single glyph.

**Comment by cmyr** [comment]
Yes this issue is not about the syntax for inline contexual rules, such as in your example; I'm referring specifically to the section of the spec dealing with [GPOS Type 2 lookups](http://adobe-type-tools.github.io/afdko/OpenTypeFeatureFileSpecification.html#6b-gpos-lookuptype-2-pair-adjustment-positioning). (I'll update the original issue to clarify this, thanks)

**Comment by frankrolf** [comment]
Sorry, I didn’t want to seem disingenuous. My thinking is the usual notation
```
pos A A -100;
```
actually is a semantic exception, which stands for
```
pos A -100 A;
```
or 
```
pos A' -100 A;
```
(the `'` being unnecessary because the context is clear)

This concept can be easier understood in longer contexts, such as
```
pos H A' -100 A;
```
